File: kms-samples-python/cmk_encrypt.py
import argparse
import base64
import json

from aliyunsdkcore.client import AcsClient
from aliyunsdkkms.request.v20160120 import EncryptRequest


def kms_encrypt(client, plaintext, key_alias):
    request = EncryptRequest.EncryptRequest()
    request.set_accept_format('JSON')
    request.set_KeyId(key_alias)
    request.set_Plaintext(plaintext)
    response = json.loads(client.do_action(request))
    return response.get('CiphertextBlob')


def read_text_file(in_file):
    with open(in_file, 'r') as f:
        content = f.read()
    return content


def write_text_file(out_file, content):
    with open(out_file, 'w') as f:
        f.write(content)


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('--ak', help='the access key id')
    parser.add_argument('--as', help='the access key secret')
    parser.add_argument('--region', default='cn-hangzhou', help='the region id')
    args = vars(parser.parse_args())

    client = AcsClient(args["ak"], args["as"], args["region"])
    # client.set_verify(False)

    key_alias = 'alias/Apollo/WorkKey'
    in_file = './certs/key.pem'
    out_file = './certs/key.pem.cipher'

    # Read private key file in text mode
    in_content = read_text_file(in_file)

    # Encrypt
    cipher_text = kms_encrypt(client, base64.b64encode(in_content.encode('utf-8')), key_alias)

    # Write encrypted key file in text mode
    write_text_file(out_file, cipher_text)


if __name__ == '__main__':
    main()

File: kms-samples-go/cmk_encrypt_best_practices/cmk_encrypt.go
package main

import (
	"encoding/base64"
	"fmt"
	"io/ioutil"
	"log"
	"os"

	"github.com/aliyun/alibaba-cloud-sdk-go/services/kms"
)

const (
	AccessKeyId     = "AccessKeyId"
	AccessKeySecret = "AccessKeySecret"
	RegionId        = "cn-hangzhou"
)

func kmsEncrypt(client *kms.Client, plaintext, keyAlias string) (string, error) {
	request := kms.CreateEncryptRequest()
	request.Scheme = "https"
	request.KeyId = keyAlias
	request.Plaintext = plaintext
	response, err := client.Encrypt(request)
	if err != nil {
		return "", fmt.Errorf("encrypt error:%v", err)
	}
	return response.CiphertextBlob, nil
}

func main() {
	client, err := kms.NewClientWithAccessKey(RegionId, os.Getenv(AccessKeyId), os.Getenv(AccessKeySecret))
	if err != nil {
		log.Fatalf("NewClientWithAccessKey error:%+v\n", err)
	}

	keyAlias := "alias/Apollo/WorkKey"
	inFile := "./certs/key.pem"
	outFile := "./certs/key.pem.cipher"

	//Read private key file in text mode
	inContent, err := ioutil.ReadFile(inFile)
	if err != nil {
		log.Fatalf("ioutil.ReadFile error:%+v\n", err)
	}

	//Encrypt
	base64Content := base64.StdEncoding.EncodeToString(inContent)
	cipherText, err := kmsEncrypt(client, base64Content, keyAlias)
	if err != nil {
		log.Fatalf("kmsEncrypt error:%+v\n", err)
	}

	//Write encrypted key file in text mode
	err = ioutil.WriteFile(outFile, []byte(cipherText), 0644)
	if err != nil {
		log.Fatalf("ioutil.WriteFile error:%+v\n", err)
	}
}
